src/main/java/com/ea/restaurant/utils/Oauth2Util.java

```diff
@@ -0,0 +1,60 @@
+package com.ea.restaurant.utils;
+
+import com.ea.restaurant.entities.AppClient;
+import com.ea.restaurant.entities.AppClientScope;
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.JWEHeader;
+import com.nimbusds.jose.JWEObject;
+import com.nimbusds.jose.Payload;
+import com.nimbusds.jose.crypto.DirectEncrypter;
+import com.nimbusds.jwt.JWTClaimsSet;
+import java.util.Date;
+
+public final class Oauth2Util {
```

Comment:
Looks like there is a lot of repetitive logic in these methods could you optimize it?

Reply:
Done.